src/librustc/hir/intravisit.rs

```diff
@@ -718,11 +718,11 @@ pub fn walk_ty_param_bound<'v, V: Visitor<'v>>(visitor: &mut V, bound: &'v TyPar
 
 pub fn walk_generic_param<'v, V: Visitor<'v>>(visitor: &mut V, param: &'v GenericParam) {
     match *param {
-        GenericParam::Lifetime(ref ld) => {
+        GenericParameter::Lifetime(ref ld) => {
```

Comment:
General comment: we prefer the shorter `Param` to `Parameter` (changes should be from the latter to the former, *not* the other way around).